### DISCUSSION TITLE
Invoke actions are being excuted after state change

### DISCUSSION THREAD

**Discussion by goldenratio** [statelyai/xstate#3122] [Q&A] [answered]
Hi,

Here is a code snippet https://xstate.js.org/viz/?gist=a067f6436b515620d8f26d1679a80c8e 

- I would like the end state to be 'gameOver' instead of 'idle'
- When state machine reached 'initialized' state, lives value inside context is not up-to date. Because of this state doesn't work as I expected.
- If I use delay inside 'initialized' - it works as expected.

Any ideas how to resolve this? I would like to avoid delay.

**Comment by Andarist** [comment, marked as answer]
To update context please use `assign` - after I've made this change the end state was correctly chosen as `gameOver`.

**Reply by goldenratio** [reply]
thank you!
